### PR TITLE
Add signature verification on endpoints and test code

### PR DESCRIPTION
**Pull request by TrustHenry**

### CODE REVIEW

File: packages/relay/src/routers/DefaultRouter.ts [hunk 1]
```diff
@@ -289,7 +307,24 @@ export class DefaultRouter {
             const signature: string = String(req.body.signature); // 서명
 
             // TODO amount > 0 조건 검사
-            // TODO 서명검증
+            // 서명검증
+            const userNonce = (await this.getLedgerContract()).nonceOf(signer);
```
Comment (Kang-Simon) [inline packages/relay/src/routers/DefaultRouter.ts:311]: const userNonce = await (await this.getLedgerContract()).nonceOf(signer);

Comment (Kang-Simon) [inline packages/relay/src/routers/DefaultRouter.ts:311]:
```suggestion
            const userNonce = await (await this.getLedgerContract()).nonceOf(signer);
```

File: packages/relay/src/routers/DefaultRouter.ts [hunk 2]
```diff
@@ -289,7 +307,24 @@ export class DefaultRouter {
...
+            if (
+                !ContractUtils.verifyPayment(
+                    purchaseId,
+                    amount,
+                    email,
+                    franchiseeId,
+                    signer,
+                    await userNonce,
```
Comment (Kang-Simon) [inline packages/relay/src/routers/DefaultRouter.ts:319]:
```suggestion
                   userNonce,
```

Comment (TrustHenry) [reply packages/relay/src/routers/DefaultRouter.ts]: Thanks for your reviews

File: packages/relay/src/routers/DefaultRouter.ts [hunk 3]
```diff
@@ -399,7 +421,14 @@ export class DefaultRouter {
             const signature: string = String(req.body.signature); // 서명
 
             // TODO amountMileage > 0 조건 검사
-            // TODO 서명검증
+            // 서명검증
+            const userNonce = (await this.getLedgerContract()).nonceOf(signer);
+            if (!ContractUtils.verifyExchange(signer, email, amountMileage, await userNonce, signature))
```
Comment (Kang-Simon) [inline packages/relay/src/routers/DefaultRouter.ts:426]:
```suggestion
            const userNonce = await (await this.getLedgerContract()).nonceOf(signer);
            if (!ContractUtils.verifyExchange(signer, email, amountMileage, userNonce, signature))
```

Comment (TrustHenry) [reply packages/relay/src/routers/DefaultRouter.ts:426]: 적용했습니다.

**Review by Kang-Simon — APPROVED** [review]
LGTM

**Review by MichaelKim20 — APPROVED** [review]
LGTM

**Pull request merged**